include/poggles/texture.h

```diff
@@ -12,36 +15,30 @@ namespace poggles
 class POGGLES_EXPORT texture
 {
 public:
-  explicit texture(GLenum target);
-  explicit operator GLuint() const
-  {
-    return static_cast<GLuint>(m_texture_handle.value());
-  }
+  explicit texture(GLenum in_target);
+
+  explicit operator GLuint() const;
+
   static void activate(GLenum texture_num) { glActiveTexture(texture_num); }
-  void bind(GLenum texture_num = GL_TEXTURE0) const
-  {
-    activate(texture_num);
-    glBindTexture(m_original_target,
-                  static_cast<GLuint>(m_texture_handle.value()));
-  }
-
-  // TODO more general form for these functions
-  void load(GLenum target, std::string const& filename, GLint level = 0);
-  void load(GLenum target,
-            unsigned char*,
-            int width,
-            int height,
-            int channels,
-            GLint level = 0);
-  void load(GLenum target,
-            float*,
-            int width,
-            int height,
-            int channels,
-            GLint level = 0);
+
+  void bind(GLenum texture_num = GL_TEXTURE0) const;
+
+  GLenum target;
 
 private:
-  GLenum m_original_target;
   texture_handle m_texture_handle;
 };
-}  // namespace poggles
+
+auto loadFromFile(texture& tex,
+                  std::filesystem::path const& filepath,
```

Comment:
I mildly prefer to use `poggles::texture::fromFile` as the name for these, but its a minor preference